### PR TITLE
Spatial_oM: Add support for cellular shape profiles

### LINKED ISSUE #1479: Structure_oM: Add support for Cellular section properties
#### Description:

Support for Cellular beams to be added.

Could either be done as a special shape profile, or as a new ISectionProperty.

Benefits of former is keeping Sections by material as currently done.

Benefits of latter is separation of section shape values and other properties. Can add a separate class to handle the opening shapes and sizes.

### PR DESCRIPTION
**Pull request by IsakNaslundBh**
### Issues addressed by this PR

Closes #1479

### Test files

https://burohappold.sharepoint.com/:f:/s/BHoM/Egum0nEYkI9LkPQ6ETB3SDYBd5H2x0SZhXBS9DVdz7K-Kg?e=9KCZ4i

### CODE REVIEW

**Review by peterjamesnugent — CHANGES_REQUESTED** [review]
Happy with the introduction with the `ISteelSection` and `VoidedISectionProfile` - we may want to consider what it's called given we refer to openings/void on the descriptions.

File: Spatial_oM/ShapeProfiles/VoidedISectionProfile.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        [Length]
+        [Description("Height of the T section at each end of the opening.")]
+        public virtual double HeightTSection { get; }
+
+        [Length]
+        [Description("Full width of both flanges between the extreme fibres of the flanges.")]
+        public virtual double Width { get; }
+
+        [Length]
+        public virtual double WebThickness { get; }
```
[the remark this answers is not in the document]
Comment (IsakNaslundBh) [reply Spatial_oM/ShapeProfiles/VoidedISectionProfile.cs:60]: kind of intentional (or copied over the ISectionProfile, and modified it). Not sure what to say other than "The thickness of the web", and not sure what that adds really?

File: Spatial_oM/ShapeProfiles/CellularOpenings/HexagonalOpening.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        [Description("Height of the hexagonal part of the opening, equal to the height of the cut being made in the section being castelated.\n" +
+                     "For case of non-zero SpacerHeight, the total height of the opening will be Height+SpacerHeight.")]
+        public virtual double Height { get; }
+
+        [Length]
+        [Description("Total width of the opening.")]
+        public virtual double Width { get; }
+
+        [Length]
+        [Description("Length of the intermidiate web post as well as top of the opening.")]
```
[the remark this answers is not in the document]
Comment (IsakNaslundBh) [reply Spatial_oM/ShapeProfiles/CellularOpenings/HexagonalOpening.cs:48]: -->        ___
           /      \
            \___/

referring to that part of the opening, like, the hoirisontal part of the opening, and it in not jsut top, but top and bottom, really.

Agree not really clear.

"Length of the intermediate web post. Same as length of horizontal top and bottom edges of the opening"
?

Happy for any better wording though...

**Review by peterjamesnugent — DISMISSED** [review]
Agree with changes made and all my previous comments have been resovled.

**Comment by IsakNaslundBh** [comment]
@BHoMBot check compliance

**Comment by IsakNaslundBh** [comment]
The compliance failure is due to all `IProfile` types having an additional "ShapeType" parameter that is hardcoded to the profile.
If this is to be fixed, all shapeprofiles needs to be updated, 

Issue for handling of this raised https://github.com/BHoM/BHoM/issues/1488

**Comment by IsakNaslundBh** [comment]
@BHoMBot this is a DevOps instruction. I am authorising dispensation to be granted on check ref. 11910920934

**Comment by IsakNaslundBh** [comment]
@BHoMBot check core
@BHoMBot check serialisation

**Comment by IsakNaslundBh** [comment]
@BHoMBot this is a DevOps instruction. I am authorising dispensation to be granted on check ref. 11912375834

**Review by peterjamesnugent — APPROVED** [review]
Reapprove.

**Comment by IsakNaslundBh** [comment]
@BHoMBot check versioning

**Comment by IsakNaslundBh** [comment]
@BHoMBot check installer

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: versioning, ready-to-merge

**Comment by FraserGreenroyd** [comment]
@BHoMBot this is a DevOps instruction. I am requesting neutral checks on: installer

**Pull request merged**